client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,366 @@
+		self.label_file = self.builder.get_object('SFTPClientGUI.label')
+		col = Gtk.TreeViewColumn('Transfer')
+		col_text = Gtk.CellRendererText()
+		col.pack_start(col_text, True)
+		col.add_attribute(col_text, "text", 0)
+		self.treeview_transfer.append_column(col)
+		self._tvmodel = Gtk.ListStore(str)
+		self.treeview_transfer.connect('size-allocate', self._treeview_changed)
+		self.treeview_transfer.set_model(self._tvmodel)
+		self.treeview_transfer.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
+		menu_item = Gtk.MenuItem.new_with_label('Clear')
+		menu_item.connect('activate', self.log_clear)
+		self.popup_menu.append(menu_item)
+		self.popup_menu.show_all()
+
+	def _signal_treeview_button_pressed(self, _, event):
+		if event.button == Gdk.BUTTON_SECONDARY:
+			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
+			return True
+		return
+
+	def log_generic(self, string):
+		self._tvmodel.prepend((string,))
+
+	def log_clear(self, _):
+		self._tvmodel.clear()
+
+	def _treeview_changed(self, widget, event, data=None):
+		adj = self.scroll.get_vadjustment()
+		adj.set_value(0)
+
+	def log_transfer_upload(self, path, size, is_folder, error=False):
+		if error:
+			string = 'Error uploading ' + path
+		elif is_folder:
+			string = 'Uploading folder ' + path + ' and all its children...'
+		else:
+			string = 'Uploading ' + path
+		self._tvmodel.prepend((string,))
+
+class DirectoryBase(object):
+	def __init__(self, treeview, logger):
+		col = Gtk.TreeViewColumn('Files')
+		self.logger = logger
+		self.treeview_local = treeview
+		col_text = Gtk.CellRendererText()
+		col_img = Gtk.CellRendererPixbuf()
+		col.pack_start(col_img, False)
+		col.pack_start(col_text, True)
+		col.add_attribute(col_text, 'text', 0)
+		col.add_attribute(col_img, 'pixbuf', 1)
+		col.set_sort_column_id(0)
+		col_perm = Gtk.TreeViewColumn('Permissions')
+		col_perm.pack_start(col_text, True)
+		col_perm.add_attribute(col_text, 'text', 3)
+		col_size = Gtk.TreeViewColumn('Size')
+		col_size.pack_start(col_text, True)
+		col_size.add_attribute(col_text, 'text', 4)
+		col_size.set_sort_column_id(5)
+		col_date = Gtk.TreeViewColumn('Date Modified')
+		col_date.pack_start(col_text, True)
+		col_date.add_attribute(col_text, 'text', 6)
+		col_date.set_sort_column_id(6)
+		self.treeview_local.append_column(col)
+		self.treeview_local.append_column(col_perm)
+		self.treeview_local.append_column(col_size)
+		self.treeview_local.append_column(col_date)
+
+		self.treeview_local.connect('row-expanded', self.signal_expand_row)
+		self.treeview_local.connect('row-collapsed', self.signal_collapse_row)
+		self._tvmodel = Gtk.TreeStore(str, Pixbuf, str, str, str, int, str)
+
+		self.local_hidden = True
+		self._get_popup_menu()
+
+	def _get_popup_menu(self):
+		self.treeview_local.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.popup_menu = Gtk.Menu.new()
```

Comment:
It would be really nice if there were also an option in this menu to collapse all entries in the treeview.